packages/web/src/features/common-space/frames/CommonSpaceInfoFrame/CommonSpaceInfoSecondFrame.tsx

```diff
@@ -0,0 +1,161 @@
+      value: "2",
+      selectable: true,
+    },
+    {
+      label: "다용도실 (학부 학생회관 별관 3104호-3105호)",
+      value: "3",
+      selectable: false,
+    },
+    {
+      label: "체조실 (스포츠 콤플렉스 101호)",
+      value: "4",
+      selectable: true,
+    },
+    {
+      label: "제1무예실 (스포츠 컴플렉스 102호)",
+      value: "5",
+      selectable: true,
+    },
+    {
+      label: "제2무예실 (스포츠 컴플렉스 103호)",
+      value: "6",
+      selectable: true,
+    },
+    {
+      label: "제3무예실 (스포츠 컴플렉스 106호)",
+      value: "7",
+      selectable: true,
+    },
+  ];
+
+  // TODO: 이름 전화번호 동아리 목록 백에서 받아오기
+
+  const [selectedValue, setSelectedValue] = useState("");
+  const [hasSelectError, setHasSelectError] = useState(false);
+  const [dateTimeRange, setDateTimeRange] = useState<[Date, Date]>();
+
+  useEffect(() => {
+    const allConditionsMet =
+      Boolean(selectedValue) && !hasSelectError && !!dateTimeRange;
+    setNextEnabled(allConditionsMet);
+  }, [selectedValue, hasSelectError, setNextEnabled, dateTimeRange]);
+
+  useEffect(() => {
+    setCommonSpace({
+      ...commonSpace,
+      space:
+        mockCommonSpaceList.find(item => item.value === selectedValue)?.label ||
+        "",
+    });
+  }, [selectedValue, setCommonSpace]);
+
+  const diffHours =
+    dateTimeRange && differenceInHours(dateTimeRange[1], dateTimeRange[0]);
+  const diffMinutes =
+    dateTimeRange && differenceInMinutes(dateTimeRange[1], dateTimeRange[0]);
+
+  useEffect(() => {
+    if (dateTimeRange) {
+      setCommonSpace(prev => ({
+        ...prev,
+        reservation: {
+          start: dateTimeRange[0],
+          end: dateTimeRange[1],
+        },
+      }));
+    }
+  }, [dateTimeRange, setCommonSpace]);
+
+  return (
+    <>
+      <Select
+        items={mockCommonSpaceList}
+        selectedValue={selectedValue}
+        onSelect={setSelectedValue}
+        label="공용공간"
+        setErrorStatus={setHasSelectError}
+      />
+      {commonSpace.space ? (
+        <>
+          <Info
+            text={`${commonSpace.space}는 하루에 최대 4시간, 일주일에 최대 10시간 사용할 수 있습니다.`}
+          />
+          <StyledCard type="outline">
+            <Timetable
+              data={Array(7 * 48).fill(false)}
+              setDateTimeRange={setDateTimeRange}
+            />
+            <StyledCardOuter>
+              {dateTimeRange && (
+                <StyledInfoCard type="outline">
+                  <Typography type="p">선택 시간</Typography>
+                  <Typography type="p">
+                    {format(dateTimeRange[0], "M/d(E) ", { locale: ko })}
+                    {format(dateTimeRange[0], "HH:mm", { locale: ko })} ~
+                    {format(dateTimeRange[1], "HH:mm", { locale: ko })} (
+                    {`${diffHours}시간`}
+                    {diffMinutes! % 60 ? ` ${diffMinutes! % 60}분` : ""})
+                  </Typography>
+                </StyledInfoCard>
+              )}
+            </StyledCardOuter>
+          </StyledCard>
+        </>
+      ) : null}
```

Comment:
없을 때 null이면 이렇게도 많이 쓰는 것 같더라구
```suggestion
      {commonSpace.space &&
        <>
          <Info
            text={`${commonSpace.space}는 하루에 최대 4시간, 일주일에 최대 10시간 사용할 수 있습니다.`}
          />
          <StyledCard type="outline">
            <Timetable
              data={Array(7 * 48).fill(false)}
              setDateTimeRange={setDateTimeRange}
            />
            <StyledCardOuter>
              {dateTimeRange && (
                <StyledInfoCard type="outline">
                  <Typography type="p">선택 시간</Typography>
                  <Typography type="p">
                    {format(dateTimeRange[0], "M/d(E) ", { locale: ko })}
                    {format(dateTimeRange[0], "HH:mm", { locale: ko })} ~
                    {format(dateTimeRange[1], "HH:mm", { locale: ko })} (
                    {`${diffHours}시간`}
                    {diffMinutes! % 60 ? ` ${diffMinutes! % 60}분` : ""})
                  </Typography>
                </StyledInfoCard>
              )}
            </StyledCardOuter>
          </StyledCard>
        </>
      }
```